cob_mimic/src/mimic_node.cpp

```diff
@@ -324,6 +345,23 @@ class Mimic
         }
         return true;
     }
+
+    void diagnostics_timer_thread()
+    {
+        while(ros::ok())
+        {
+          ros::Duration(1.0).sleep();
+          diagnostic_updater_.update();
+        }
+    }
+
+    void produce_diagnostics(diagnostic_updater::DiagnosticStatusWrapper &stat)
+    {
+        stat.summary(diagnostic_msgs::DiagnosticStatus::OK, "Mimic running");
+        stat.add("Action is active", action_active_);
+        stat.add("Service is active", service_active_);
+        stat.add("Current goal", active_mimic_);
```

Comment:
I would replace `Current goal` with `Active mimic` then...as it could also be the default mimic or the mimic requested by service...